python/sglang/srt/layers/moe/fused_moe_triton/layer.py

```diff
@@ -205,19 +224,45 @@ def forward_cpu(
         correction_bias: Optional[torch.Tensor] = None,
         activation: str = "silu",
     ) -> torch.Tensor:
-        return moe_forward_native(
-            layer,
-            x,
-            use_grouped_topk,
-            top_k,
-            router_logits,
-            renormalize,
-            topk_group,
-            num_expert_group,
-            custom_routing_function,
-            correction_bias,
-            activation,
-        )
+        assert activation == "silu", f"activation = {activation} is not supported."
+
+        if support_amx():
+            topk_weights, topk_ids = select_experts(
+                hidden_states=x,
+                router_logits=router_logits,
+                use_grouped_topk=use_grouped_topk,
+                top_k=top_k,
+                renormalize=renormalize,
+                topk_group=topk_group,
+                num_expert_group=num_expert_group,
+                custom_routing_function=custom_routing_function,
+                correction_bias=correction_bias,
+                torch_native=True,
+            )
+
+            return fused_experts_cpu(
+                x,
+                layer.w13_weight,
+                layer.w2_weight,
+                topk_weights,
+                topk_ids,
+                False,  # inplace
```

Comment:
can we use inplace? forward_cuda uses "inplace=True"

https://github.com/sgl-project/sglang/blob/00f42707eaddfc2c0528e5b1e0094025c640b7a0/python/sglang/srt/layers/moe/fused_moe_triton/layer.py#L163

Reply:
Changed inplace to True in https://github.com/mingfeima/sglang/pull/8/commits/f09e7b12792c328c3f085c8a1bc8ddc080fd4e3c and confirmed that the generated sentence is correct.